OpenTap.Plugins.Ssh/SshResource.cs

```diff
@@ -129,5 +142,113 @@ public override string ToString()
             if (IsSession) return _step.GetFormattedName();
             return base.ToString() + $"({Connection.Username}@{Connection.Host}:{Connection.Port})";
         }
+
+        // <summary>
+        // Starts a process with the given command in the background
+        // The process PID is stored in BackgroundProcesses to keep track of all processes started by this SshResource
+        // NOTE: this only works in Linux systems
+        // </summary>
+        public Pid StartBackgroundProcess(string command)
+        {
+            // nohup allows the command to run even when the SSH session ends
+            // output is suppressed
+            // the PID of the last background process '$!' is printed to output
+            SshCommand sshCmd = SshClient.CreateCommand($"nohup {command} > /dev/null 2>&1 & echo $!");
+            string cmdOut = sshCmd.Execute();
+            
+            Log.Debug("Running command: " + sshCmd.CommandText);
+
+            Pid pid = new Pid(cmdOut);
+            BackgroundProcesses.Add(pid);
+            return pid;
+        }
+
+        // <summary>
+        // Checks if background process with the given pid is running
+        // This only works if the process was started within the same session of this SshResource
+        // NOTE: this only works in Linux systems
+        // </summary>
+        public bool IsBackgroundProcessRunning(Pid pid)
+        {
+            if (!BackgroundProcesses.Contains(pid))
+            {
+                Log.Debug($"PID={pid} was not started in this session or it is not alive anymore");
+                return false;
+            }
+
+            SshCommand command = SshClient.CreateCommand($"ps -p {pid}");
+            command.Execute();
+
+            if (command.ExitStatus == 0)
+            {
+                Log.Debug($"Process with PID={pid} is alive");
+                return true;
+            }
+
+            Log.Debug($"Process with PID={pid} is not alive. Removing it from the list of background processes");
+            BackgroundProcesses.Remove(pid);
+            return false;
+        }
+
+        // <summary>
+        // Kills a background process
+        // This only works if the process was started within the same session of this SshResource
+        // NOTE: this only works in Linux systems
+        // </summary>
+        public bool KillBackgroundProcess(Pid pid)
+        {
+            if (!IsBackgroundProcessRunning(pid))
+            {
+                Log.Debug($"No process with PID={pid} to be killed");
+                return false;
+            }
+
+            SshCommand command = SshClient.CreateCommand($"kill {pid}");
+            command.Execute();
+
+            if (command.ExitStatus == 0)
+            {
+                Log.Debug($"Gracefully killed PID={pid}");
+                BackgroundProcesses.Remove(pid);
+                return true;
+            }
```

Comment:
This exit status doesn't check if the background process exited. It checks if the `kill` command successfully sent the `SIGTERM` signal to the process. If the process ignores the signal, or doesn't handle it (because it's hanging for example) then `kill` will still return 0 without exiting the process